awsService/src/main/java/com/kovan/app/service/S3Service.java

```diff
@@ -0,0 +1,269 @@
+package com.kovan.app.service;
+
+import com.kovan.entity.Document;
+import com.kovan.exception.FileException;
+import com.kovan.service.DocumentService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.*;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import static io.micrometer.common.util.StringUtils.isBlank;
+import static java.util.UUID.randomUUID;
+
+@Service
+@Slf4j
+public class S3Service {
+
+    @Value("${bucketName}")
+    private String bucketName;
+
+    private final S3Client s3Client;
+    private final DocumentService service;
+
+    public S3Service(S3Client s3Client, DocumentService service) {
+        this.s3Client = s3Client;
+        this.service = service;
+    }
+    public String createBucket(String bucketName) {
+        try {
+            CreateBucketRequest createBucketRequest = CreateBucketRequest.builder()
+                    .bucket(bucketName)
+                    .build();
+            s3Client.createBucket(createBucketRequest);
+            log.info("Bucket {} created successfully", bucketName);
+            return "Bucket created successfully: " + bucketName;
+        } catch (S3Exception e) {
+            log.error("Error creating bucket {}: {}", bucketName, e.getMessage());
+            return "Error creating bucket: " + e.getMessage();
+        }
+    }
+
+    public String deleteBucket(String bucketName) {
+        try {
+            deleteAllObjects(bucketName);
+            DeleteBucketRequest deleteBucketRequest = DeleteBucketRequest.builder()
+                    .bucket(bucketName)
+                    .build();
+            s3Client.deleteBucket(deleteBucketRequest);
+
+            log.info("Bucket {} deleted successfully", bucketName);
+            return "Bucket deleted successfully: " + bucketName;
+        } catch (S3Exception e) {
+            log.error("Error deleting bucket {}: {}", bucketName, e.getMessage());
+            return "Error deleting bucket: " + e.getMessage();
+        }
+    }
+    public void deleteAllObjects(String bucketName) {
+    ListObjectsV2Request listObjectsV2Request = ListObjectsV2Request.builder()
+            .bucket(bucketName)
+            .build();
+    List<CompletableFuture<Void>> objectDeletionFutures = s3Client.listObjectsV2Paginator(listObjectsV2Request).stream()
+            .flatMap(response -> response.contents().stream())
+            .map(s3Object -> CompletableFuture.runAsync(() -> {
```

Comment:
This methodology gives lacks control over concurrency, which could become problematic as the number of objects and versions grows. use control the creation of thread pool by you not from ForkJoinPool. modify it everywhere wherever you are using this methodology

Reply:
Modified this methodology of creating thread pool from ForkJoinPool.